firebase-firestore/src/test/java/com/google/firebase/firestore/remote/BloomFilterTest.java

```diff
@@ -0,0 +1,169 @@
+  }
+
+  @Test
+  public void testNonEmptyBloomFilterThrowException() {
+    IllegalArgumentException negativePaddingException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], -1, 1));
+    assertThat(negativePaddingException).hasMessageThat().contains("Invalid padding: -1");
+    IllegalArgumentException overflowPaddingException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], 8, 1));
+    assertThat(overflowPaddingException).hasMessageThat().contains("Invalid padding: 8");
+
+    IllegalArgumentException negativeHashCountException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], 1, -1));
+    assertThat(negativeHashCountException).hasMessageThat().contains("Invalid hash count: -1");
+    IllegalArgumentException zeroHashCountException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], 1, 0));
+    assertThat(zeroHashCountException).hasMessageThat().contains("Invalid hash count: 0");
+  }
+
+  @Test
+  public void testBloomFilterProcessNonStandardCharacters() {
+    // A non-empty BloomFilter object with 1 insertion : "ÀÒ∑"
+    BloomFilter bloomFilter = new BloomFilter(new byte[] {(byte) 237, 5}, 5, 8);
+    assertTrue(bloomFilter.mightContain("ÀÒ∑"));
+    assertFalse(bloomFilter.mightContain("Ò∑À"));
+  }
+
+  @Test
+  public void testEmptyBloomFilterMightContainAlwaysReturnFalse() {
+    BloomFilter bloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertFalse(bloomFilter.mightContain("abc"));
+  }
+
+  @Test
+  public void testBloomFilterMightContainOnEmptyStringAlwaysReturnFalse() {
+    BloomFilter emptyBloomFilter = new BloomFilter(new byte[0], 0, 0);
+    BloomFilter nonEmptyBloomFilter =
+        new BloomFilter(new byte[] {(byte) 255, (byte) 255, (byte) 255}, 1, 16);
+
+    assertFalse(emptyBloomFilter.mightContain(""));
+    assertFalse(nonEmptyBloomFilter.mightContain(""));
+  }
+
+  /**
+   * Golden tests are generated by backend based on inserting n number of document paths into a
+   * bloom filter.
+   *
+   * <p>Full document path is generated by concatenating documentPrefix and number n, eg,
+   * projects/project-1/databases/database-1/documents/coll/doc12.
+   *
+   * <p>The test result is generated by checking the membership of documents from documentPrefix+0
+   * to documentPrefix+2n. The membership results from 0 to n is expected to be true, and the
+   * membership results from n to 2n is expected to be false with some false positive results.
+   */
+  @Test
+  @SuppressWarnings("DefaultCharset")
+  public void testBloomFilterGoldenTest() throws Exception {
+    String documentPrefix = "projects/project-1/databases/database-1/documents/coll/doc";
+
+    // Import the golden test files for bloom filter
+    HashMap<String, JSONObject> parsedSpecFiles = new HashMap<>();
+    File jsonDir = new File("src/test/resources/bloom_filter_golden_test_data");
```

Comment:
The golden files should be loaded from Android "resources", like how the spec test json files are.

The spec test json files are located in https://github.com/firebase/firebase-android-sdk/tree/master/firebase-firestore/src/test/resources/json. The directory name "json" is entirely arbitrary, so choose a different directory name for your golden test files.

Reply:
The test cases are under Android "resources", under a directory named "bloom_filter_golden_test_data".

Reply:
Okay I see that this is how the spec tests work too. Please ignore my comment. It was in error.